### PR TITLE
Expose ROI Pooling into Python API.

### PR DESCRIPTION
**Pull request by dyning**
- Expose ROI Pooling into Python API.
- Add unit testing.

### CODE REVIEW

**Comment by chengduoZH** [comment]
Please sign the CLA and make CI passed first.

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -3759,3 +3760,44 @@ def label_smooth(label,
...
+        argmaxes (Variable): Argmaxes corresponding to indices in input with shape (num_rois, channels, pooled_h, pooled_w). 
+
+    Examples:
+        .. code-block:: python
+             # assuming we have input x_feas, rois x_rois, pooled_height ph, pooled_width pw and spatial_scale scale.
+             pool_out, argmaxes = fluid.layers.roi_pool(input=x_feas, rois=x_rois, pooled_height=ph, pooled_width=pw, spatial_scale=scale)
+    """
+    helper = LayerHelper('roi_pool', **locals())
+    dtype = helper.input_dtype()
+    pool_out = helper.create_tmp_variable(dtype)
```
Comment (qingqing01) [inline python/paddle/fluid/layers/nn.py]: Also need create variable for `Argmax`.

File: python/paddle/fluid/tests/unittests/test_layers.py [hunk 2]
```diff
@@ -359,6 +359,17 @@ def test_topk(self):
             self.assertIsNotNone(indices)
         print(str(program))
 
+    def test_roi_pool(self):
+        print("test_roi_pool")
```
Comment (qingqing01) [inline python/paddle/fluid/tests/unittests/test_layers.py]: Remove this line.

File: python/paddle/fluid/layers/nn.py [hunk 3]
```diff
@@ -3759,3 +3760,44 @@ def label_smooth(label,
         outputs={"Out": smooth_label},
         attrs={"epsilon": float(epsilon)})
     return smooth_label
+
+
+def roi_pool(input, rois, pooled_height=1, pooled_width=1, spatial_scale=1.0):
```
Comment (qingqing01) [inline python/paddle/fluid/layers/nn.py]: Please also add `roi_pool` in doc : https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/api/layers.rst#nn

File: python/paddle/fluid/layers/nn.py [hunk 4]
```diff
@@ -3759,3 +3760,44 @@ def label_smooth(label,
...
+        inputs={"X": input,
+                "ROIs": rois},
+        outputs={"Out": pool_out,
+                 "Argmax": argmaxes},
+        attrs={
+            "pooled_height": pooled_height,
+            "pooled_width": pooled_width,
+            "spatial_scale": spatial_scale
+        })
+    return pool_out, argmaxes
```
Comment (wanghaoshuang) [inline python/paddle/fluid/layers/nn.py]: 一般`ROI pool` 之后的op需要`argmaxes`么？如果不需要，就不用return了吧。

File: python/paddle/fluid/tests/unittests/test_layers.py (hunk 2 above)
Comment (dyning) [reply python/paddle/fluid/tests/unittests/test_layers.py]: ok

File: python/paddle/fluid/layers/nn.py (hunk 4 above)
Comment (dyning) [reply python/paddle/fluid/layers/nn.py]: 不需要的

**Pull request closed without merging**